sections/dashboard/TrendingSynths/utils.ts

```diff
@@ -1,13 +1,13 @@
+import { Synth } from '@synthetixio/contracts-interface';
+import { wei, WeiSource } from '@synthetixio/wei';
 import { CurrencyKey } from 'constants/currency';
-import { Synth } from 'lib/synthetix';
 import { Query } from 'react-query';
-import { NumericValue, toBigNumber } from 'utils/formatters/number';
 
 // The function turns react query cached queries into a map (with the currencyKey) as the key to be used in the sorting.
 export const toCurrencyKeyMap = (
 	query: Query<unknown, unknown>[],
 	dataField?: string
-): Record<CurrencyKey, number> =>
+): Partial<Record<CurrencyKey, number>> =>
```

Comment:
why Partial here? is it because `CurrencyKey` is a defined list? It should just be a string right?. Or is it becaue it might be an empty object?

Reply:
normally `Record` expects to be a complete accounting of the type if it has limited values. As `CurrencyKey` has been converted into a set of valid strings, and this function only returns a subset of those values, `Partial` needs to be added here.